### PR TITLE
CloudFormation template for External functions

### PR DESCRIPTION
**Pull request by sfc-gh-nsharma**
--AWS cloudformation template for creating API Gateway, Lambda and Roles needed for external functions

@sf-gh-schintala - Please try the end to end flow with this template before approving the PR

### CODE REVIEW

**Comment by sfc-gh-nsharma** [comment]
@sf-gh-schintala - I've also added the Azure template to this PR. Feel free to give that a try. Also, I'm using this PR to park my changes for now till we get the documentation ready so not merging it anytime soon.

**Comment by sfc-gh-nsharma** [comment]
Rename the azure template to BasicSetup.json

File: DeploymentTemplates/aws/BasicSetup.yaml [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+  apiGatewayName:
+    Type: "String"
+    AllowedPattern: "^[a-zA-Z0-9]+[-a-zA-Z0-9-]+[-a-zA-Z0-9]+$"
+    Default: "ext-func-api"
+    Description: "API Gateway instance name"
+
+  apiGatewayIAMRoleName:
+    Type: "String"
+    AllowedPattern: "^[a-zA-Z0-9]+[-a-zA-Z0-9-]+[-a-zA-Z0-9]+$"
+    Default: "ext-func-api-role"
```
Comment (sfc-gh-nsharma) [inline DeploymentTemplates/aws/BasicSetup.yaml:36]: We should not use default value for this.

File: DeploymentTemplates/aws/BasicSetup.yaml [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    Description: "This is the role that will be used in Snowflake API Integration"
+
+  lambdaExecutionRoleName:
+    Type: "String"
+    Default: "ext-func-lambda-role"
```
Comment (sfc-gh-nsharma) [inline DeploymentTemplates/aws/BasicSetup.yaml:41]: We should not use default value for this.

**Comment by sfc-gh-nsharma** [comment]
@sf-gh-schintala - I'm going to merge this PR now and update the readme files to point to Mark's docs once they are ready. Any concerns?

**Comment by sf-gh-schintala** [comment]
@sfc-gh-nsharma sounds good.

**Pull request merged**